src/app/dataset-view/additional-components/data-component/data.component.ts

```diff
@@ -111,10 +116,43 @@ export class DataComponent extends BaseComponent implements OnInit {
         }
     }
 
-    public addData(): void {
-        const modalRef: NgbModalRef = this.ngbModalService.open(AddDataModalComponent);
-        const modalRefInstance = modalRef.componentInstance as AddDataModalComponent;
-        modalRefInstance.datasetBasics = this.datasetBasics;
+    public addData(overviewUpdate: OverviewUpdate): void {
+        const metadata = overviewUpdate.overview.metadata;
+        if (metadata.currentPollingSource || metadata.currentPushSources.length || metadata.currentTransform) {
+            this.updateNow();
+        } else if (this.datasetBasics.kind === DatasetKind.Derivative) {
+            this.navigationService.navigateToSetTransform({
+                accountName: this.datasetBasics.owner.accountName,
+                datasetName: this.datasetBasics.name,
+            });
+        } else {
```

Comment:
What if I already have `SetTransform`? I expect to launch transformations.

Reply:
![image](https://github.com/kamu-data/kamu-web-ui/assets/51016717/7ccc8378-df03-4c38-9c6e-6793d0e20019)
Yes, we will launch transformations